init.go

```diff
@@ -33,84 +42,74 @@ func opensslInit(file string) error {
 			_OPENSSL_INIT_LOAD_CONFIG|
 			_OPENSSL_INIT_LOAD_CRYPTO_STRINGS,
 		nil); err != nil {
-		dlclose(handle)
+		close()
 		return err
 	}
 	osslHandle = handle
 	return nil
 }
 
-// loadFuncs loads and initialize the OpenSSL functions.
-// See shims.go for the complete list of supported functions.
-func loadOpenSSLFuncs(handle unsafe.Pointer) {
-	mkcgoLoad_(handle)
-	if vMajor == 1 {
-		mkcgoLoad_legacy_1(handle)
-		if vPatch == 1 {
-			mkcgoLoad_111(handle)
-		}
-	} else {
-		mkcgoLoad_111(handle)
-		mkcgoLoad_3(handle)
-	}
-}
-
 // initForCheckVersion loads and initialize only the
 // functions required in [CheckVersion].
 // It returns a function that must be called to release the resources.
 // The function leaves all the global variables in the same state as they were
 // before the call.
```

Comment:
A bit more precise about the restore behavior:

```suggestion
// It returns a close function that must be called to release the resources.
//
// This function modifies the vMajor, vMinor, and vPatch global variables as
// well as other internal global variables that facilitate OpenSSL calls.
//
// If the function succeeds, calling the close function restores the previous
// state of the global variables. If it fails, the global variables are restored
// before returning.
```